### PR TITLE
0.2.0 - Working Discord and Google OAuth2 providers

### PR DESCRIPTION
**Pull request by dudeisbrendan03**
Next PR is hopefully 0.2.0

What's happened:

- [x] Working Google OAuth2 flow
- [x] Working Discord OAuth2 flow
- [x] Working module loader
- [x] Working debug tools

What's there to do:

- [ ] Review @Superigi 's AAD OAuth2 handler
- [x] Finish returns

More crap:
- [x] Functioning Generic OAuth2 flow

### CODE REVIEW

**Review by 0xh4sh — APPROVED** [review]

**Review by dudeisbrendan03 — COMMENTED** [review]
Finishing up with generic providers and Google/Discord

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@tpcofficial/big-oauth2",
-  "version": "0.1.52",
+  "version": "0.1.65",
```
Comment (dudeisbrendan03) [inline package-lock.json:3]: 0.2.0 on PR

File: providers/discord-classic.js.disabled [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/**
```
Comment (dudeisbrendan03) [inline providers/discord-classic.js.disabled:1]: New provider structure, this is now deprecated.

File: providers/generic.js [hunk 3]
```diff
@@ -1,18 +1,110 @@
...
+        } catch (e) {
+            this.libs.log.error(`[${this.platform_name}] Failed to start OAuth2 flow: Couldn't generate (and/or) return the consent uri`);
+            throw `[${this.platform_name}] Failed to generate consent uri\n${e}`;
+        }
+    }
 
-// // Query for data (may differ across platforms)
\ No newline at end of file
+    async stopFlow(flowResponse) {//Should receive the token, automatically and prepare it for the user - the token is not stored and this should return USER DATA only
+        return new Promise (async (resolve, reject) => {
```
Comment (dudeisbrendan03) [inline providers/generic.js:44]: Shouldn't have an async function inside your promise, although if we do not we will never be able to fulfil the promise as we require it

Comment (dudeisbrendan03) [reply providers/generic.js:44]: @0xh4sh

**Review by dudeisbrendan03 — COMMENTED** [review]
Only thing that stands out is the new promise in generic.js

File: providers/generic.js (hunk 3 above)
Comment (dudeisbrendan03) [reply providers/generic.js]: Solved

**Review by 0xh4sh — APPROVED** [review]

**Pull request merged**